specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1136 @@
+
+Fabrikam app uses Contoso's Muffin and Waffle packages via Dynamic Dependencies, installing them if
+necessary. These packages are added to the package graph and later removed when no longer needed.
+
+```c#
+var DoAwesomeStuffUsingMuffinsAndWaffles()
+{
+    var packageSet = new PackageSet();
+    var muffin = new PackageSetItem();
+    muffin.PackageFamilyName = "contoso.muffin_1234567890abc";
+    muffin.MinVersion = ToVersion(1, 2, 3, 4);
+    muffin.PackageUri = new Uri("c:\\contoso\\muffin-1.2.3.4.msix");
+    packageSet.Add(muffin);
+    var waffle = new PackageSetItem();
+    waffle.PackageFamilyName = "contoso.waffle_1234567890abc";
+    waffle.MinVersion = ToVersion(2, 4, 6, 8);
+    waffle.PackageUri = new Uri("https://contoso.com/waffle-2.4.6.8.msix");
+    packageSet.Add(waffle);
+
+    var packageDeploymentManager = PackageDeploymentManager.GetDefault();
+    var options = new EnsureIsReadyOptions();
+    var deploymentResult = await packageDeploymentManager.EnsurePackageSetIsReadyAsync(packageSet, options);
+    try
+    {
+        if (deplymentResult.Status == PackageDeploymentStatus.CompletedSuccess)
+        {
+            Console.WriteLine("OK");
+        }
+        else
+        {
+            Console.WriteLine("Error {}", deploymentResult.ExtendedError());
+        }
+    }
+    catch (Exception ex)
+    {
+        Console.WriteLine(ex.ToString());
+    }
+
+    var packageRuntimeManager = PackageRuntimeManager.GetDefault();
+    var packageSetRuntimeDisposition = packageRuntimeManager.AddPackageSet(packageSet);
+    DoAwesomeStuff();
+    packageRuntimeManager.RemovePackageSet(packageSetRuntimeDisposition);
+}
+
+PackageVersion ToVersion(uint major, uint minor, uint build, uint revision)
+{
+    var version = new PackageVersion();
+    if ((major > UInt16.MaxValue) ||
+        (minor > UInt16.MaxValue) ||
+        (build  > UInt16.MaxValue) ||
+        (revision > UInt16.MaxValue))
+    {
+        throw new ArgumentOutOfRangeException();
+    }
+    version.major = (ushort) major;
+    version.minor = (ushort) minor;
+    version.build = (ushort) build;
+    version.revision = (ushort) revision;
+    return version;
+}
+```
+
+**NOTE:** This differs from the PackageRuntimeManager.AddPackageSet()() example by explicitly
+removing the packages dynamically added to the package graph when no longer needed.
+
+## 4.6. PackageVolume.Repair()
+
+Fabrikam app checks if a PackageVolume is OK and if not, prompts the user to confirm it should
+proceed to repair the PackageVolume.
+
+```c#
+var CheckAndFixPackageVolume(string path)
+{
+    var packageVolume = PackageVolumeManager.FindPackageVolumeByPath(path);
+    if (!packageVolume.VerifyIsOk())
+    {
+        bool ok = PromptUserForConfirmation();
+        if (!ok)
+        {
+            return;
+        }
+    }
+    try
+    {
+        packageVolume.Repair();
+    }
+    catch (Exception ex)
+    {
+        Console.WriteLine(ex.ToString());
+    }
+}
```

Comment:
try-catch block should be inside "if" clause?  If volume is OK then there's no need to repair.